02-04-domain-validations/internal/platform/server/handler/courses/create.go

```diff
@@ -24,7 +25,14 @@ func CreateHandler(courseRepository mooc.CourseRepository) gin.HandlerFunc {
 
 		course, err := mooc.NewCourse(req.ID, req.Name, req.Duration)
 		if err != nil {
-			ctx.JSON(http.StatusBadRequest, err.Error())
+			if errors.Is(err, mooc.ErrEmptyCourseName) {
+				ctx.JSON(http.StatusBadRequest, err.Error())
+			} else if errors.Is(err, mooc.ErrShortCourseName) {
+				ctx.JSON(http.StatusLengthRequired, err.Error())
+			} else {
+				ctx.JSON(http.StatusBadRequest, err.Error())
+			}
+
```

Comment:
It's not directly related with the course's contents itself, but I think you can return a `400 Bad Request` for all of them. IMHO, the `411 Length Required` was defined for a more specific case (missing `Content-Length` header), as stated on the [HTTP RFC](https://tools.ietf.org/html/rfc7230).

Reply:
You are right about 411 status code. I just decided return a different status code depending the error code received, it's what I understood when I read the exercise but surely I understood badly.